FlowCal/excel_ui.py

```diff
@@ -773,17 +781,29 @@ def generate_histograms_table(samples_table, samples):
             if pd.notnull(samples_table[header][sample_id]):
                 # Get units in which bins are being reported
                 unit = samples_table[header][sample_id]
-                # Store bins
-                bins = sample.domain(channel)
+                # Decide whether to produce histograms in linear or log scale
+                log = (unit != 'Channel') and \
+                    bool(sample.amplification_type(channel)[0])
+                # Define number of bins
+                nbins = min(sample.resolution(channel), max_bins)
+                # Calculate bin edges
+                bin_edges = sample.hist_bins(channel, nbins, log)
+                # Calculate bin centers
+                if log:
+                    bin_centers = (np.log10(bin_edges[:-1]) + \
+                        np.log10(bin_edges[1:]))/2
+                    bin_centers = 10**bin_centers
+                else:
+                    bin_centers = (bin_edges[:-1] + bin_edges[1:])/2
+                # Store bin centers
                 hist_table.loc[(sample_id,
                                 channel,
-                                'Bin Values ({})'.format(unit)),
-                               columns[0:len(bins)]] = bins
+                                'Bin Centers ({})'.format(unit)),
+                                columns[0:len(bin_centers)]] = bin_centers
                 # Calculate and store histogram counts
-                bin_edges = sample.hist_bin_edges(channel)
                 hist, __ = np.histogram(sample[:,channel], bins=bin_edges)
                 hist_table.loc[(sample_id, channel, 'Counts'),
-                               columns[0:len(bins)]] = hist
+                               columns[0:len(bin_centers)]] = hist
```

Comment:
ok. this off-by-one # of bin edges relative to # of counts is only relevant for the `Histograms` tab, right? if so, I don't have strong feelings about it; "bin centers" is fine by me